inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -701,10 +700,52 @@ std::shared_ptr<ngraph::Node> V10Parser::XmlDeserializer::createNode(
         if (!ngraphNode) {
             THROW_IE_EXCEPTION << "Opset " << params.version << " doesn't contain the operation with type: " << type;
         }
+        if (type == "Constant") {
+            auto constant = ngraph::as_type_ptr<ngraph::op::v0::Constant>(ngraphNode);
+            constant->need_alloc_buffer_on_visit_attrubutes(false);
+        }
         ngraphNode->set_arguments(inputs);
         XmlDeserializer visitor(node, weights, opsets);
-        if (ngraphNode->visit_attributes(visitor))
+        if (ngraphNode->visit_attributes(visitor)) {
+            if (type == "Constant") {
+                auto constant = ngraph::as_type_ptr<ngraph::op::v0::Constant>(ngraphNode);
+
+                pugi::xml_node dn = node.child("data");
+
+                if (dn.empty())
```

Comment:
I don't understand why do we need so much constant specific code here?
From my point of view we have a visit attributes method for such things.

Reply:
This because we don't have  everything in one place. 

At XmlDeserializer::on_adapter(const std::string& name, ngraph::ValueAccessor<void*>& adapter) we don't have a pointer to Constant node. adapter  is not  allocated as  well, as we going to use shared buffer, so size and shape of AlignedBuffer not  available.

At  XmlDeserializer::createNode we  don't have  info about shape, size, offset, except from xml_node.

Reply:
The `XmlDeserializer` has a pointer to weights, `m_data` is a shared pointer to AlignedBuffer. I think `XMLDeserializer` can allocate the buffer for the constant op